orderer/common/multichannel/registrar.go

```diff
@@ -1044,3 +1057,15 @@ func (r *Registrar) removeSystemChannel() error {
 
 	return nil
 }
+
+func (r *Registrar) removeLedgerAsync(channelID string) {
+	go func() {
+		r.pendingRemoval.Store(channelID, true)
+		if err := r.ledgerFactory.Remove(channelID); err != nil {
+			r.pendingRemoval.Store(channelID, false)
+			logger.Errorf("Ledger factory failed to remove empty ledger '%s', error: %s", channelID, err)
+			return
+		}
+		r.pendingRemoval.Delete(channelID)
```

Comment:
just take the lock, remove, and unlock:
```
r.lock.Lock()
delete(r.pendingRemoval,channelID)
r.lock,Unlock()
```

or you can define this 3 liner as a defer in the entrance to the function you give to the goroutine.

Reply:
wrapped in the lock